### ISSUE TITLE
Weird behavior with  autocomplete

### ISSUE THREAD

**Issue by rafuru** [nolimits4web/Framework7#934]
Hi!

I have a hardtyped list of objects and I'm using an autocomplete to select an item. 

The problem is I'm using valueproperty and textproperty parameters to load additional data on select, this is my source:

**data.js**

```
var emisores = [
    {
        id:1,
        razonSocial: "Max Payne",
        country:"México"
    },
  {
        id:2,
        razonSocial: "Wonder Red",
        country:"México"
    }
];
```

*_myapp.js
*_

```
var autoCompleteEmisores = myApp.autocomplete({
    input: '#pdEmisor',
    openIn: 'dropdown',    
    expandInput: true,
    valueProperty:'id',
    textProperty: 'razonSocial', //object's "text" property name
    source: function (autocomplete, query, render) {
        var results = [];
        if (query.length === 0) {
            render(results);
            return;
        }
        for (var i = 0; i < emisores.length; i++) {
            if (emisores[i].razonSocial.toLowerCase().indexOf(query) >= 0){
               results.push(emisores[i]);
            }
        }
        render(results);
    },
    onChange:function(autocomplete,val){
        alert(val.razonSocial);
    }
});
```

When I select an item I get an "undefined" as value, but if I set "razonSocial" as valueProperty i get the entire object... which is not intended , I mean.. yes, is the info I need.. but It shouldn't be happening.

**Comment by ZanderBrown** [comment]
Could you provide your demonstration as a jsFiddle / CodePen please?

**Comment by nolimits4web** [comment]
Fixed by #1566
